cli/lsp/language_server.rs

```diff
@@ -894,9 +901,10 @@ impl Inner {
       return Ok(None);
     }
     let line_index = self.get_line_index_sync(&specifier).unwrap();
-    let file_diagnostics: Vec<&Diagnostic> = self
+    let file_diagnostics: Vec<Diagnostic> = self
       .diagnostics
       .diagnostics_for(&specifier, &DiagnosticSource::TypeScript)
+      .cloned()
```

Comment:
This is to work around `self` being borrowed both mutably and immutably in this method. Ditto the change to `add_ts_fix_action()`.